### ISSUE TITLE
[Bug]: Precision issue with flaggems AttentionFLBackend integration in vllm-plugin-FL

### ISSUE THREAD

**Issue by tsingmicro-public-e** [flagos-ai/vllm-plugin-FL#166]
### Bug Description

When using the AttentionFLBackend from flaggems via vllm-plugin-FL, there is a significant precision issue in the generated output. The model produces repetitive tokens and incorrect factual descriptions (e.g., repeating the abbreviation "沪" instead of using "申").

Environment
vllm-plugin-FL Version: v0.1.0

FlagGems Version: 4.2.1rc0

Hardware: NVIDIA H20

CUDA Version: 12.6 (NVCC 12.6.85)

Model: Qwen3.5-35B 

### Steps to Reproduce

Steps to Reproduce
Modify vllm-plugin-FL/vllm_fl/dispatch/backends/flaggems/flaggems.py to force the use of AttentionFLBackend:

`def attention_backend(self, use_mla: bool = False) -> str: `
    `...`
    `# Forcing the return of AttentionFLBackend`
    ` return "vllm_fl.dispatch.backends.flaggems.impl.attention.AttentionFLBackend"`

2. Run inference with the prompt: '帮我介绍一下上海市?' (Please introduce Shanghai).
Expected Behavior
The model should generate accurate and coherent text.
Example of normal output:

Generated text: \n\n<think>\n\n</think>\n\n上海，简称“沪”或“申”，是中国的直辖市，也是国家中心城市、国际经济、金融、贸易、

### Expected Behavior

The abbreviation is incorrectly repeated as “沪”或“沪” (should be “沪”或“申”).

The tokenization or attention weights seem to drift, leading to degraded generation quality compared to the vendor's default implementation.

<img width="1140" height="496" alt="Image" src="https://github.com/user-attachments/assets/daf49f0e-938c-4c93-97b9-53800180ea6c" />

### Actual Behavior

The output contains repetitive characters and incorrect formatting.
Example of abnormal output:

Generated text: \n\n<think>\n\n</think>\n\n上海（Shanghai），简称“沪”或“沪”，是**中国最大的城市**，也是**国际经济、金融、

<img width="1116" height="634" alt="Image" src="https://github.com/user-attachments/assets/bca84d16-a135-4063-bc65-baeeb2de1350" />

### Environment

- vllm-plugin-FL version: v0.1.0
- vLLM version: v0.13.0
- Python version: 3.12.3
- OS: Ubuntu 24.04.1 LTS
- Hardware (GPU/accelerator): H20
- FlagGems version (if applicable):  4.2.1rc0
- FlagCX version (if applicable): 未使用

**Comment by BIGKUANG** [comment]
`
from vllm import LLM, SamplingParams
import torch
import os
import time



PROMPT_LIST = [
    "帮我介绍一下上海市?"
] * 16


MODEL_PATH = "/data/test_infer/models/Qwen3.5-35B-A3B/"

if __name__ == "__main__":
    prompts = PROMPT_LIST

    # 1. 采样参数调整：保持 temperature=0，并显式传入 seed
    sampling_params = SamplingParams(
        max_tokens=20, 
        temperature=0.0,  # 强制贪婪解码，消除采样随机性
        seed=42           # 显式固定采样种子
    )

    # 2. LLM 初始化：保持你的 seed=42
    llm = LLM(
        model=MODEL_PATH,
        enforce_eager=True,
        trust_remote_code=True,
        enable_chunked_prefill=True,
        served_model_name="Qwen3.5-35b",
        tensor_parallel_size=16,
        max_model_len=8192,  
        max_num_seqs=16,
        seed=42           # 固定模型内部算子和分布式的种子
    )

    # Generate texts from the prompts.
    outputs = llm.generate(prompts, sampling_params)
    for idx, output in enumerate(outputs):
        prompt = output.prompt
        generated_text = output.outputs[0].text
        token_ids = output.outputs[0].token_ids
        print(f"[{idx}] Prompt: {prompt!r}")
        print(f"[{idx}] Generated text: {generated_text!r}")
        print(f"[{idx}] Token IDs: {token_ids}")

    print("Inference end...")
    time.sleep(2)
    print("Exit success...")
`

**Comment by 0x45f** [comment]
try fix: https://github.com/flagos-ai/FlagGems/pull/3410
